### ISSUE TITLE
Wrong parameter list wrapping

### ISSUE THREAD

**Issue by vRallev** [pinterest/ktlint#1119]
## Expected Behavior
IntelliJ formats a top-level function like that:
```kotlin
private fun <T, U : LayerRendering> StatelessWorkflow<
  AuthenticatorInput, AuthenticatorOutput, LayeredScreen<PosLayering>
  >.RenderContext.renderChildToLayeredScreen(
  workflow: Workflow<T, Unit, U>,
  props: T
): LayeredScreen<MainAndModal> {
  ...
}
```

## Observed Behavior
KtLint complains about:
```
file.kt:4:3: Unexpected indentation (expected 4, actual 2) (parameter-list-wrapping)
file.kt:5:3: Unexpected indentation (expected 4, actual 2) (parameter-list-wrapping)
file.kt:6:1: Unexpected indentation (expected 2, actual 0) (parameter-list-wrapping)
```

Changing the function to:
```kotlin
private fun <T, U : LayerRendering> StatelessWorkflow<
  AuthenticatorInput, AuthenticatorOutput, LayeredScreen<PosLayering>
  >.RenderContext.renderChildToLayeredScreen(
    workflow: Workflow<T, Unit, U>,
    props: T
  ): LayeredScreen<MainAndModal> {
  ...
}
```
will trigger these errors:
```
file.kt:4:1: Unexpected indentation (4) (should be 2) (indent)
file.kt:5:1: Unexpected indentation (4) (should be 2) (indent)
file.kt:6:1: Unexpected indentation (2) (should be 0) (indent)
```

## Steps to Reproduce
See sample above.

## Your Environment

* Version of ktlint used: 0.41.0
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): custom Gradle task
* Operating System and version: Mac / Linux
* `indent_size = 2`

**Comment by ileasile** [comment]
Maybe the same bug:

`build.gradle.kts`:
```kotlin
tasks.test {
    systemProperties = mutableMapOf(
        "junit.jupiter.displayname.generator.default" to
            "org.junit.jupiter.api.DisplayNameGenerator\$ReplaceUnderscores",

        "junit.jupiter.execution.parallel.enabled" to
            doParallelTesting.toString() as Any,
        "junit.jupiter.execution.parallel.mode.default" to
            "concurrent",
        "junit.jupiter.execution.parallel.mode.classes.default" to
            "concurrent"
    )
}
```

`ktlintFormat` fails with:
Task :ktlintKotlinScriptFormat FAILED
```
C:\Projects\kotlin-jupyter\build.gradle.kts:178:9 Unexpected indentation (expected 4, actual 8)
C:\Projects\kotlin-jupyter\build.gradle.kts:181:5 Unexpected indentation (expected 4, actual 8)
C:\Projects\kotlin-jupyter\build.gradle.kts:183:1 Unexpected indentation (expected 4, actual 8)
C:\Projects\kotlin-jupyter\build.gradle.kts:184:27 Unexpected indentation (expected 4, actual 8)
```

Works fine for 0.40.0, bug was introduced in 0.41.0

**Comment by bethcutler** [comment]
Another example (likely the same issue):
```
class Outer {
  class ExceptionallyParameterizedClass<
    ExceptionallyLongParameter1,
    ExceptionallyLongParameter2,
    ExceptionallyLongParameter3,
    ExceptionallyLongParameter4>
  private constructor(
    private val firstArgument: Type<ExceptionallyLongParameter1>,
    private val secondArgument: Type<ExceptionallyLongParameter3>
  ) {
    // foobar
  }
}
```

This gives `Unexpected indentation (expected 2, actual 4) (parameter-list-wrapping)` errors on both the parameter lines for the constructor, as well as the constructor's closing paren.

However, a similar example:
```
class Outer {
  class ExceptionallyParameterizedClass<ExceptionallyLongParameter1, ExceptionallyLongParameter2>
  private constructor(
    private val firstArgument: Type<ExceptionallyLongParameter1>,
    private val secondArgument: Type<ExceptionallyLongParameter3>
  ) {
    // foobar
  }
}
```

has no errors.
